doc/ch-image.rst

```diff
@@ -927,6 +928,156 @@ trading off the time needed to retrieve an image from cache.
+   3. I also deliberately left out an option to change the output compression
+      algorithm, under the assumption that the default is good enough. This
+      can be revisited later IMO if needed.
+
+Image formats
+-------------
+
+:code:`export` knows about these values of :code:`FMT`:
+
+  :code:`ch-image`
+    Internal storage for Charliecloud's unprivileged image builder (Dockerfile
+    interpreter) :code:`ch-image`.
+
+  :code:`dir`
+    Ordinary filesystem directory (i.e., not a mount point) containing an
+    unpacked image. Output directories that already exist are replaced if they
+    look like an image; otherwise, exit with an error.
+
+  :code:`squash`
+    SquashFS filesystem archive containing the flattened image. SquashFS
+    archives are much like tar archives but are mountable, including by
+    :code:`ch-run`'s internal SquashFUSE mounting. Most systems have at least
+    the SquashFS-Tools installed which allows unpacking into a directory, just
+    like tar. Due to this greater flexibility, SquashFS is preferred to tar.
+
+    **Note:** Conversions to and from SquashFS are quite noisy due to the
+    verbosity of the underlying :code:`mksquashfs(1)` and
+    :code:`unsquashfs(1)` tools.
+
+  :code:`tar`
+    Tar archive containing the flattened image with no layer sub-archives.
+    Output tarballs are always gzipped and must end in :code:`.tar.gz`;
+    input tarballs can be any compression acceptable to :code:`tar(1)`.
+
+All of these are local formats; :code:`export` does not know how to push
+or pull images.
+
+
+Format inference
+----------------
+
+:code:`export` tries to save typing by guessing formats when they are
+reasonably clear. This is done against filenames, rather than file contents,
+so the rules are the same for output descriptors that do not yet exist.
+
+Format inference is done for both :code:`IN` and :code:`OUT`. The first
+matching glob below yields the inferred format. Paths need not exist in the
+filesystem.
+
+  1. :code:`*.sqfs`, :code:`*.squash`, :code:`*.squashfs`: SquashFS.
+
+  2. :code:`*.tar`, :code:`*.t?z`, :code:`*.tar.?`, :code:`*.tar.??`: Tarball.
+
+  3. :code:`/*`, :code:`./*`, i.e. absolute path or relative path with
+     explicit dot: Directory.
+
+  4. If `ch-image` is installed: :code:`ch-image` internal storage.
+
+  5. Otherwise: No format inference.
+
+
+Examples
+--------
+
+Typical build-to-run sequence for image :code:`foo/bar` using :code:`ch-run`'s
+internal SquashFUSE code, inferring the output format::
+
+  $ ch-image build -t hello .
+  [...]
+  $ ch-image export hello /var/tmp/hello.sqfs
+  input:   ch-image  hello
+  output:  squashfs  /var/tmp/hello.sqfs
+  copying ...
+  done
+  $ ch-run /var/tmp/hello.sqfs -- echo hello
+  hello
+
+Same conversion, but no format inference::
+
+  $ ch-image export -o squash hello /var/tmp/foobar.sqfs
+  input:   ch-image  hello
+  output:  squashfs  /var/tmp/hello.sqfs
+  copying ...
+  done
+
```

Comment:
Ah, we will want to update this to also include an example covering @reidpr's suggest supplemental build argument, `--output`, e.g., `$ch-image build --output [...]`.